### DISCUSSION TITLE
:Rocks search {rock-name}

### DISCUSSION THREAD

**Discussion by thepragmaticmero** [lumen-oss/rocks.nvim#541] [Q&A] [answered]
`cargo search` exists O-O
is `:Rocks search ` gonna be implemented? Idk where to find exact package names.

**Comment by mrcjkb** [comment, marked as answer]
We have fuzzy tab-completion for `:Rocks install`, which should be equivalent?

**Reply by thepragmaticmero** [reply]
on I'll try fuzzy finding then, I'm used to a search feature in most package managers. Fuzzy finding sounds nice. Also, Is it necessary to install another package manager? Your wiki says that it has lazy.nvim support, but I'm trying to keep things minimal

**Reply by mrcjkb** [reply]
We don't support lazy.nvim.
We have an (optional) external module, called rocks-lazy.nvim, which adds lazy-loading capabilities to rocks.nvim.

> Is it necessary to install another package manager

rocks.nvim currently uses luarocks as a backend.
We're working on [rocks](https://github.com/nvim-neorocks/rocks.nvim), which we plan on replacing luarocks with.
Think of it as similar to how LSP works. You have a minimal client written in lua (rocks.nvim) and an external program that does the heavy lifting. 
That way, you can benefit from parallel execution and multi threading (which you don't get with pure Lua implementations).
This will likely be more noticeable when our own rocks backend is ready.

See

- #539

**Reply by thepragmaticmero** [reply]
Thanks for the clarification. I'll be on the lookout :D
